packages/legacy/core/App/components/network/NetInfo.tsx

```diff
@@ -12,63 +11,57 @@ const NetInfo: React.FC = () => {
   const { t } = useTranslation()
   const [hasShown, setHasShown] = useState(false)
 
+  const showNetworkWarning = useCallback(() => {
+    setHasShown(true)
+    Toast.show({
+      type: ToastType.Error,
+      autoHide: true,
+      text1: t('NetInfo.NoInternetConnectionTitle'),
+    })
+  }, [t])
+
+  // will be null until network state is known
   const isConnected = silentAssertConnectedNetwork()
 
   useEffect(() => {
-    const _showNetworkWarning = () => {
-      setHasShown(true)
-      Toast.show({
-        type: ToastType.Error,
-        autoHide: true,
-        text1: t('NetInfo.NoInternetConnectionTitle'),
-      })
-    }
-    // Network is available, do further testing according to CFG.disableMediatorCheck
-    if (!disableMediatorCheck) {
-      // Network is available
-      if (isConnected) {
-        // Check mediator socket, also assert internet reachable
-        assertMediatorReachable().then((status) => {
-          if (status) {
-            Toast.hide()
-            return
-          } else {
-            // Network is available but cannot access nediator, display toast
-            _showNetworkWarning()
-          }
-        })
-        return
-      } else if (!hasShown) {
-        _showNetworkWarning()
+    // Only check general internet connection if mediator check is disabled
+    if (disableMediatorCheck) {
```

Comment:
I though a change was implemented to make the check generic because some teams don't use a mediator?

Reply:
That hadn't been done yet, but I've added that to this PR now